src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -77,12 +77,16 @@ public static Prefix[] getPrefixesPresent(ArgumentMultimap argMultimap) {
      * @throws ParseException if the user input does not conform the expected format
      */
     public static Patient createPatient(ArgumentMultimap argMultimap, Prefix[] prefixes) throws ParseException {
+        assert prefixes.length > 1;
         // filling the fields with default values
+        for (Prefix r : REQUIRED_PREFIXES) {
+            assert argMultimap.getValue(r).isPresent();
+        }
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
         Phone phone = new Phone(Phone.getDefaultPhone());
         Email email = new Email(Email.getDefaultEmail());
         Gender gender = new Gender(Gender.getDefaultGender());
-        IcNumber icNumber = new IcNumber(IcNumber.getDefaultIcNumber());
+        IcNumber icNumber = new IcNumber(argMultimap.getValue(PREFIX_IC_NUMBER).get());
```

Comment:
Yup agree with the change here since IC Number is a required prefix just like name, and cannot be subject to a random default value